tests/python/pants_test/pants_run_integration_test.py

```diff
@@ -383,6 +356,7 @@ def run_pants_with_workdir_without_waiting(self, command, workdir, config=None,
       env = os.environ.copy()
     if extra_env:
       env.update(extra_env)
+    env.update(PYTHONPATH=os.pathsep.join(sys.path))
```

Comment:
Hey gang: I believe that this code might have caused an issue with "loose-source" plugins not ending up "all the way" on the python path when running integration tests.

The loose-source plugins are loaded with `[GLOBAL] pythonpath`, but the subprocessed pants runs fail with:
```
Exception caught: (pants.base.exceptions.BackendConfigurationError)
  File "/opt/twitter_mde/package/eepython36/68cc93925092a6fc31d254392558fa8c840f7bd5407db9435ca48ff1b8f6c07f/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/twitter_mde/package/eepython36/68cc93925092a6fc31d254392558fa8c840f7bd5407db9435ca48ff1b8f6c07f/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/__main__.py", line 8, in <module>
    pants_loader.main()
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/bin/pants_loader.py", line 78, in main
    PantsLoader.run()
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/bin/pants_loader.py", line 74, in run
    cls.load_and_execute(entrypoint)
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/bin/pants_loader.py", line 67, in load_and_execute
    entrypoint_main()
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/bin/pants_exe.py", line 32, in main
    PantsRunner(start_time=start_time).run()
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/bin/pants_runner.py", line 85, in run
    options_bootstrapper=options_bootstrapper
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/bin/local_pants_runner.py", line 142, in create
    options_bootstrapper=options_bootstrapper,
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/bin/local_pants_runner.py", line 91, in parse_options
    build_config = BuildConfigInitializer.get(options_bootstrapper)
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/init/options_initializer.py", line 38, in get
    cls._cached_build_config = cls(options_bootstrapper).setup()
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/init/options_initializer.py", line 69, in setup
    self._bootstrap_options.backend_packages
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/init/options_initializer.py", line 58, in _load_plugins
    return load_backends_and_plugins(plugins, working_set, backend_packages)
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/init/extension_loader.py", line 32, in load_backends_and_plugins
    load_build_configuration_from_source(build_configuration, backends)
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/init/extension_loader.py", line 113, in load_build_configuration_from_source
    load_backend(build_configuration, backend_package)
  File "/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/init/extension_loader.py", line 129, in load_backend
    raise BackendConfigurationError(f'Failed to load the {backend_module} backend: {e!r}')

Exception message: Failed to load the twitter.pants_internal.register backend: ModuleNotFoundError("No module named 'twitter.pants_internal.tasks.some_twitter_task'",)
```

Which is odd, because it sounds like maybe the `register.py` is located, but not some of the other code...

Reply:
Nothing comes to mind. Perhaps you can create a repro against pants own loose source plugins in a local IT? Easier, just hand-edit `/Users/stuhood/src/source/.pants.d/pyprep/sources/cf3f7a2a3d17140e6302e01caf2b23a165031273/pants/init/extension_loader.py` to print out `sys.path` and dump `sys.modules` to start.

Reply:
Holy mackerel. This was undeclared BUILD level dependencies in these tests. That is very unexpected, and very welcome... I have to go change our README to indicate that integration tests _do_ need validate dependency lists. Exciting =)